File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cpp

class Solution {
public:
  int lengthOfLongestSubstring(string s) {
    int n = s.size();
    int l = 0, r = 0, ans = 0;
    map<char,int> frq;
    while(r < n) {
       if(not frq[s[r]]) {
        frq[s[r++]]++;
       }
       else frq[s[l++]]--;
       ans = max(ans, r - l);
    }
    return ans;
  }
};

File: Data Structures & Algorithms/custom-sort-string/submission-2.cs
public class Solution
{
    public string CustomSortString(string order, string s)
    {
        StringBuilder ans = new StringBuilder();
        int[] frq = new int[26];

        for (int i = 0; i < s.Length; i++)
        {
            frq[s[i] - 'a']++;
        }

        for (int i = 0; i < order.Length; i++)
        {
            while (frq[order[i] - 'a'] > 0)
            {
                ans.Append(order[i]);
                frq[order[i] - 'a']--;
            }
        }

        for (int i = 0; i < 26; i++)
        {
            while (frq[i] > 0)
            {
                char ch = (char)(i + 'a');
                ans.Append(ch);
                frq[i]--;
            }
        }

        return ans.ToString();
    }
}